src/client/js/app.js

```diff
@@ -18,10 +18,32 @@
  */
 import {instance as router} from './router';
 
+function updateOnlineStatus(event) {
+  document.body.classList.toggle('offline', !navigator.onLine);
+  document.querySelectorAll('.add-to-cart').forEach((btn) => {
```

Comment:
It would be good to have a generic pattern for this. Maybe an `.online-only` class for buttons so that it can be added to anything that requires a connection to work.

Reply:
I'm not convinced - e.g. some buttons you might want to disable, some to hide, and others to change behaviour (attach different listener). I'm rather against premature generalisation - let's have the use case and then think how to optimise it?

Reply:
Fair. I think what I'm uncomfortable with here is centralising all the offline functionality here adding a tight coupling between all offline behaviour and this file. What I was suggesting is similar to the  [convention MDC web follows](https://material.io/develop/web/components/input-controls/line-ripple/), e.g. `<div class="mdc-line-ripple"></div>` to indicate an element will receive the ripple behaviour. Though, I'm happy to deal with that once we have multiple use cases.